sklearn/tree/_export.py

```diff
@@ -683,19 +687,24 @@ def export(self, decision_tree, ax=None):
     def recurse(self, node, tree, ax, max_x, max_y, depth=0):
         import matplotlib.pyplot as plt
 
+        # kwargs for annotations without a bounding box
+        non_box_kwargs = dict(
+            zorder=100 - 10 * depth,
+            xycoords="axes fraction",
+        )
+        if self.fontsize is not None:
+            non_box_kwargs["fontsize"] = self.fontsize
+
+        # kwargs for annotations with a bounding box
         kwargs = dict(
-            bbox=self.bbox_args.copy(),
             ha="center",
             va="center",
-            zorder=100 - 10 * depth,
-            xycoords="axes fraction",
+            bbox=self.bbox_args.copy(),
             arrowprops=self.arrow_args.copy(),
+            **non_box_kwargs,
```

Comment:
If `kwargs` applies to annotations with a bounding box, then it's strange to have `non_box_kwargs` here.

If they share some kwargs, then I prefer this naming:

```suggestion
            **common_box_kwargs,
```